include/ccf/crypto/jwk.h

```diff
@@ -96,4 +118,19 @@ namespace crypto
   };
   DECLARE_JSON_TYPE_WITH_BASE(JsonWebKeyRSAPrivate, JsonWebKeyRSAPublic);
   DECLARE_JSON_REQUIRED_FIELDS(JsonWebKeyRSAPrivate, d, p, q, dp, dq, qi);
+
+  struct JsonWebKeyEdDSAPublic : JsonWebKey
```

Comment:
Both EdDSA and ECDH has the same kty (key type) `OKP` and the same jwk format.
So, we have a few options about how to define OKP JWK structs:
1. Use the current implementation. When we want to ECDH support, we can add `JsonWebKeyECDHPublic` separately (with exact same contents.) I think it makes sense because they are for different key algorithms even though the JWK format is shared.
2. Define `JsonWebKeyOKPPublic` and share it for both EdDSA and ECDH. It also makes sense because they use the same JWK format. There is a downside for CCF developers (not for users). Our code sometimes chosoe functions to call based on type like https://github.com/microsoft/CCF/blob/7683f3e93b8f7eb9de1763362f5fb8059d3e018a/src/js/crypto.cpp#L342 . If a type is related to multiple algorithms (EdDSA and ECDH here), the code would be less simple because we would need a code to check the algorithm.
3. Make `JsonWebKeyOKPPublic` a parent class for classes for `JsonWebKeyEdDSAPublic ` and `JsonWebKeyECDHPublic`. I can't imagine a case where it is useful though.


Please let me know what you think.

Reply:
I think 1. is fine.